src/RMPI/helpers.jl

```diff
@@ -95,37 +97,14 @@ function Rimu.sort_into_targets!(dtarget::MPIData, ws::NTuple{NT,W}, statss) whe
     end
     sort_into_targets!(dtarget,lwm) # combine walkers from different MPI ranks
     stats = sum(statss) # combine stats from all threads
-    MPI.Allreduce!(stats, +, dtarget.comm) # add stats from all MPI ranks
-    return dtarget, ws, stats
+    res_stats = _communicate_stats(stats, dtarget.comm) # add stats from all MPI ranks
+    return dtarget, ws, res_stats
 end
 function Rimu.sort_into_targets!(dtarget::MPIData, w::AbstractDVec, stats)
     # single threaded MPI version
     sort_into_targets!(dtarget,w) # combine walkers from different MPI ranks
-    vstats = convert(Vector,stats) # necessary for MPI.Allreduce
-    MPI.Allreduce!(vstats, +, dtarget.comm) # add stats from all MPI ranks
-    return dtarget, w, vstats
-end
-
-function Rimu.sort_into_targets!(target, source, ::Type{P}, s::DistributeStrategy) where P
-    # now target is just a local data structure, e.g. DVec
-    # allocate local buffers for sorting
-    bufs = [Vector{P}(undef,length(source)) for i in 1:(s.np-1)] # type-stable
-    lens = zeros(Int,(s.np-1))
-    # sort source into send buffers
-    @inbounds for (key,val) in pairs(source)
-        tr = targetrank(key, s.np)
-        if tr < s.id
-            lens[tr+1] +=1
-            bufs[tr+1][lens[tr+1]] = key => val
-        elseif tr > s.id
-            lens[tr] +=1
-            bufs[tr][lens[tr]] = key => val
-        else # tr == s.id
-            target[key] += val # local: just copy to target
-        end
-    end
-    # call strategy-specific method (with 5 arguments):
-    return sort_into_targets!(target, bufs, lens, P, s)
```

Comment:
Hmm, I don't understand currently why this code was deleted. Is it not necessary to do the sorting into buffers according to `targetrank()` in order for `mpi_point_to_point` to work?

Reply:
I moved this part to `onesided.jl` because it was only used there. Point to point uses preallocated buffers to do this.